components/profile/create/Modal.vue

```diff
@@ -121,4 +125,51 @@ const handleFormSubmition = async (profileData: ProfileFormData) => {
     console.error(error)
   }
 }
+
+const onSelectFarcaster = () => {
+  if (farcasterUserData.value) {
+    stage.value = 3
+    useFarcaster.value = true
+  } else {
+    farcasterSignInIsInProgress.value = true
+    loginWithFarcaster().then(() => {
+      farcasterSignInIsInProgress.value = false
+      stage.value = 3
+      useFarcaster.value = true
+    })
+  }
+}
+
+const OnSelectStartNew = () => {
+  stage.value = 3
+  useFarcaster.value = false
+}
+
+const loginWithFarcaster = async () => {
+  const channel = await createChannel()
+
+  if (channel?.data?.url) {
+    // Open a new tab with the URL
+    const farcasterTab = window.open(channel.data.url, '_blank')
+    const userData = await appClient.watchStatus({
+      channelToken: channel.data.channelToken,
+      timeout: 60_000,
+      interval: 1_000,
+    })
+
+    farcasterTab.close()
+
+    if (userData?.data?.state !== 'completed') {
+      console.error('No user data found')
+      return
+    }
+    if (userData.data.nonce !== channel.data.nonce) {
+      console.error('nonce mismatch')
+      return
+    }
+    farcasterUserData.value = userData.data
+  } else {
+    console.error('URL not found in channel data')
+  }
```

Comment:
How about showing a notification toast for these errors?

Reply:
> How about showing a notification toast for these errors?

nice catch! makes sense, already created one for reminding that the action is on new tab

- https://github.com/kodadot/nft-gallery/issues/10234

Reply:
Sure

Let's do this in #10234